articles/google-play-in-app-reviews/index.md

```diff
@@ -0,0 +1,250 @@
+
+```java
+public class MainActivity extends AppCompatActivity {
+    ReviewManager reviewManager;
+    ReviewInfo reviewlnfo = null;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        getReviewInfo();
+        startReviewFlow();
+    }
+
+    private void getReviewInfo() {
+        reviewManager = ReviewManagerFactory.create(getApplicationContext());
+        Task<ReviewInfo> manager = reviewManager.requestReviewFlow();
+        manager.addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                reviewlnfo = task.getResult();
+            } else {
+                Toast.makeText(getApplicationContext(), "In App ReviewFlow failed to start", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    public void startReviewFlow() {
+        if (reviewlnfo != null) {
+            Task<Void> flow = reviewManager.launchReviewFlow(this, reviewlnfo);
+            flow.addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(Task<Void> task) {
+                    Toast.makeText(getApplicationContext(), "In App Rating complete", Toast.LENGTH_LONG).show();
+                }
+            });
+        }
+        else {
+            Toast.makeText(getApplicationContext(), "In App Rating failed", Toast.LENGTH_LONG).show();
+        }
+    }
+}
+```
+
+### Testing
+To test our application, we will use internal app sharing. To confirm that this works, add a `Button` to trigger the review flow. Remember, we are using a call to action for testing. In production applications, calls to action such as `Button` should be avoided when using this API.
+
+- Add a button in your activity.xml
+
+```xml
+<Button
+    android:id="@+id/button"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="159dp"
+    android:layout_marginEnd="164dp"
+    android:layout_marginBottom="600dp"
+    android:text="SHOW IN APP REVIEW DIALOG"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent" />
+```
+
+- Declare the button
+
+```java
+Button mButton;
+```
+
+Below `onCreate`,
+
+- Create an instance of the `mButton`
+
+```java
+mButton = findViewById(R.id.button);
+```
+
+- Add `OnClickListener` and call `startReviewFlow()`
+
+```java
+mButton.setOnClickListener(new View.OnClickListener() {
+    @Override
+    public void onClick(View view) {
+        startReviewFlow();
+    }
+});
+```
+
+#### What you need
+- A Google Play Console account.
+- An application already published in the Google Play store.
+
+#### How to test
+1. Generate a signed app bundle/APK. Note the APK signing key and the `applicationId` should be the same as the already published application.
```

Comment:
This process seems to have two parts:
1. Sharing the application using the internal testing program. This is to be done by the developer
2. The user interaction with the application
Maybe you could find a way to separate the two and since this is a technical article, give a higher concentration to the technical part of deploying as you have already explained what the application does at the coding section.

Reply:
The case of this testing is done by the developer and not the user. The developer is testing if the Review flow is working using the internal app sharing provided by the Google play console. internal app sharing provides a sharable link that helps to test any google core API, aiming to test what cannot be tested locally. It a virtually deployed app aside from the actual production app available for the users. If you have the Release apps to testing tracks permission, you’re authorized to upload APKs for internal sharing by default.